pandas/core/resample.py

```diff
@@ -2115,32 +2118,42 @@ def _adjust_dates_anchored(
     closed: Literal["right", "left"] = "right",
     origin: TimeGrouperOrigin = "start_day",
     offset: Timedelta | None = None,
+    unit="ns",
 ) -> tuple[Timestamp, Timestamp]:
     # First and last offsets should be calculated from the start day to fix an
     # error cause by resampling across multiple days when a one day period is
     # not a multiple of the frequency. See GH 8683
     # To handle frequencies that are not multiple or divisible by a day we let
     # the possibility to define a fixed origin timestamp. See GH 31809
-    first = first.as_unit("ns")
-    last = last.as_unit("ns")
+    first = first.as_unit(unit)
+    last = last.as_unit(unit)
     if offset is not None:
-        offset = offset.as_unit("ns")
-
-    origin_nanos = 0  # origin == "epoch"
+        offset = offset.as_unit(unit)
+
+    # TODO is there anything which can be reused here?
+    freq_value = freq.nanos
+    if unit == "us":
+        freq_value = freq_value // 1_000
+    elif unit == "ms":
+        freq_value = freq_value // 1_000_000
+    elif unit == "s":
+        freq_value = freq_value // 1_000_000_000
```

Comment:
@jbrockmendel is there any existing function which can be reused here?

there is `periods_per_second`, but that takes `npydatetime_unit` rather than `str` (is there a function to convert between them?)

Reply:
abbrev_to_npy_unit

in tzconversion there are two places where we do approximately this, one of them has a comment to de-duplicate